### PR TITLE
Implement State Recovery Command for interoperability - Closes #7020 and #7021 

### LINKED ISSUE #7020: Implement mainchain State Recovery Command verification
### Description

Add `state recovery command` verification to mainchain interoperability module. 

```
# The terminated account has to exist for this sidechain
if terminatedStateAccount(trs.params.chainID) does not exists or terminatedStateAccount(trs.params.chainID).initialized == False:
    return False

terminatedStateRoot = terminatedStateAccount(trs.params.chainID).stateRoot

let queryKeys and storeQueries be empty arrays

for each entry in trs.params.storeEntries:
    push entry.storeKey to queryKeys
    query = { key: entry.storeKey,
            value: SHA-256(entry.storeValue),
            bitmap: entry.bitmap}
    push query to storeQueries

proofOfInclusionStores = { siblingHashes: trs.params.siblingHashes, queries : storeQueries}

return SMTVerify(queryKeys, proofOfInclusionStores, terminatedStateRoot)
```

### Acceptance Criteria 

- Add unit tests
- Import this class for use in sidechain interoperability module

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### What was the problem?

This PR resolves #7020 and #7021

### How was it solved?

- Implemented State Recovery verification hook 🌱 
- Implemented State Recovery execution hook 🌱 
- Added tests ✔️ 

### How was it tested?

Added unit tests

### CODE REVIEW

File: framework/src/modules/interoperability/schema.ts [hunk 1]
```diff
@@ -602,3 +602,52 @@ export const registrationSignatureMessageSchema = {
 		},
 	},
 };
+
+export const stateRecoveryParams = {
```
Comment (Incede) [inline framework/src/modules/interoperability/schema.ts:606]: `stateRecoveryParamsSchema`

File: framework/src/modules/interoperability/schema.ts [hunk 2]
```diff
@@ -602,3 +602,52 @@ export const registrationSignatureMessageSchema = {
...
+	$id: '/modules/interoperability/mainchain/commands/state_recovery',
```
Comment (Incede) [inline framework/src/modules/interoperability/schema.ts:607]: Prefer camelCase - `stateRecovery`

**Review by Incede — CHANGES_REQUESTED** [review]
Unit tests for `execute` seem to be missing and few minor comments.

**Review by ishantiw — CHANGES_REQUESTED** [review]
Overall LGTM, couple of comments

File: framework/test/unit/modules/interoperability/mainchain/commands/state_recovery.spec.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+			),
+		};
+		await terminatedStateSubstore.setWithSchema(
+			chainIDAsBuffer,
+			terminatedStateAccount as any,
+			terminatedStateSchema,
+		);
+		transactionContext = createTransactionContext({
+			transaction,
+			stateStore,
```
[the remark this answers is not in the document]
Comment (mitsuaki-u) [reply framework/test/unit/modules/interoperability/mainchain/commands/state_recovery.spec.ts:112]: `createTransactionContext` was made specifically for testing purposes and should be fine to use in this test.

[the remark this answers is not in the document]
Comment (mitsuaki-u) [reply framework/test/unit/modules/interoperability/mainchain/commands/state_recovery.spec.ts:112]: As per discussion we have decided to use mock for this test as testing for sparseMerkleTree is already done extensively in the `lisk-tree` library

**Review by ishantiw — APPROVED** [review]
LGTM 🎉

**Pull request merged**